src/app/token/token_handler.rs

```diff
@@ -0,0 +1,50 @@
+use super::constants::{TOKEN_FILE_NAME, TOKEN_FOLDER_NAME};
+use anyhow::Context;
+use std::fs::{self, File};
+use std::io::Write;
+use std::path::PathBuf;
+
+pub fn get_token_file_path() -> Result<PathBuf, anyhow::Error> {
+    let home_dir = dirs::home_dir().context("failed to get home directory")?;
+    let token_dir = home_dir.join(TOKEN_FOLDER_NAME);
+    Ok(token_dir.join(TOKEN_FILE_NAME))
+}
+
+pub fn get_token() -> anyhow::Result<Option<String>> {
```

Comment:
shouldn't there be no option?
if the token is not expected to be present - why call this in the first place?

Reply:
In the scenario where the user is not logged in and the token is not present, the `get_token()` function will return `None`. It is the responsibility of the function caller to handle the `None` case.